src/identity/customca.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::CertificateProvider;
+use async_trait::async_trait;
+use kube::{
+    api::{Api, PostParams, ListParams, WatchEvent},
+    client::Client,
+};
+use tracing::{instrument, warn};
+use crate::{identity::manager::Identity, tls::SanChecker};
+use crate::identity::Error;
+use crate::tls;
+use futures::{StreamExt, TryStreamExt};
+use k8s_openapi::{api::certificates::v1::{CertificateSigningRequest, CertificateSigningRequestSpec}, ByteString};
+use std::str;
+use rand::prelude::*;
+
+#[derive(Clone)]
+pub struct CustomSigner {
+    pub signer: Option<std::string::String>,
```

Comment:
```suggestion
    pub signer: String,
```

Not sure we need option since its required?